bin/ofs/src/fuse.rs

```diff
@@ -401,26 +386,39 @@ impl PathFilesystem for Fuse {
             flags
         );
 
-        let (is_read, is_write) = self.check_flags(flags)?;
+        let (is_read, is_trunc, is_append) = self.check_flags(flags | libc::O_CREAT as u32)?;
 
         let path = PathBuf::from(parent).join(name);
         self.op
             .write(&path.to_string_lossy(), Bytes::new())
             .await
             .map_err(opendal_error2errno)?;
 
-        let metadata = Metadata::new(EntryMode::FILE);
-        let attr = metadata2file_attr(&metadata, SystemTime::now(), self.uid, self.gid);
+        let inner_writer = if is_trunc || is_append {
+            let writer = self
+                .op
+                .writer_with(&path.to_string_lossy())
+                .append(is_append)
+                .await
+                .map_err(opendal_error2errno)?;
+            Some(Arc::new(Mutex::new(InnerWriter { writer, written: 0 })))
+        } else {
+            None
+        };
+
+        let now = Utc::now();
+        let metadata = Metadata::new(EntryMode::FILE)
+            .with_last_modified(now)
+            .with_content_length(0);
+        let attr = metadata2file_attr(&metadata, now.into(), self.uid, self.gid);
```

Comment:
The file doesn't exist from the storage side.

Reply:
File created in https://github.com/apache/opendal/pull/4617/files/ccebc378ac27f986f931801353299af9a4a7f539#diff-6c6477cf532c8c42dcca74370b8416681397353768d74db6bae663a4f727bac7R392-R395

Reply:
How about calling `stat` to get the metadata instead of building via `Metadata::new()`?

Reply:
If we want to maintain attributes, it's better to build them manually rather than creating metadata and then converting it.